lib/src/facts/linux/networking_resolver.cc

```diff
@@ -71,4 +89,85 @@ namespace facter { namespace facts { namespace linux {
         return interface;
     }
 
+    void networking_resolver::read_routing_table()
+    {
+        auto ip_command = lth_exe::which("ip");
+        if (ip_command.empty()) {
+            LOG_DEBUG("Could not find the 'ip' command. Network bindings will not be populated from routing table");
+            return;
+        }
+
+        auto parse_route_line = [](string& line, std::vector<route>& routes) {
+            vector<boost::iterator_range<string::iterator>> parts;
+            boost::split(parts, line, boost::is_space(), boost::token_compress_on);
+            if (parts.size() % 2 != 1) {
+                LOG_WARNING("Could not process routing table entry: Expected a destination followed by key/value pairs, got '{}'", line);
+                return true;
+            }
+
+            route r;
+            r.destination.assign(parts[0].begin(), parts[0].end());
+            // Iterate over key/value pairs and add the ones we care
+            // about to our routes entries
+            for (size_t i = 1; i < parts.size(); i += 2) {
```

Comment:
This could be an algorithm (group by n and iterator over tuples). Candidate for a later date maybe.